test/langtools/tools/javac/options/BCPOrSystemNotSpecified.java

```diff
@@ -206,7 +209,9 @@ private void prepareBCP(Path target) throws IOException {
                     Files.createDirectories(targetDir);
                     try (InputStream in = file.openInputStream()) {
                         String sourcePath = file.getName();
-                        int sepPos = sourcePath.lastIndexOf(fileSep);
+                        // Here, we should use file system separator instead of the operating system separator.
+                        String fileSystemSep = ((PathFileObject) file).getPath().getFileSystem().getSeparator();
```

Comment:
what about using: System.getProperty("file.separator")?

Reply:
That would be incorrect.  
That system property is for the default file system, and may not be the correct separator for a user-defined file system, as is possible with `java.nio.file.FileSystem`

Reply:
@vicente-romero-oracle Explaining a bit more ... on Windows, the default file system separator is `\` but we're dealing here with `Path` objects from the "internal" JRT file system, which uses `/` as a separator. That's why we have to be careful to use the right separator.  On Unix/Linux/macOS, the separators are the same, which is how the test has passed on those platforms.

Reply:
I see thanks for the clarification